modules/derivation/src/main/scala-3/sangria/macros/derive/DeriveObjectTypeMacro.scala

```diff
@@ -0,0 +1,866 @@
+              default match {
+                case Some((defaultTpe, defaultValue)) =>
+                  defaultTpe.asType match
+                    case ('[d]) =>
+                      (
+                        Expr.summon[ToInput[d, _]],
+                        Expr.summon[FromInput[Option[i]]],
+                        Expr.summon[ArgumentType[Option[i]]]) match
+                        case (Some(toInput), Some(fromInput), Some(argType)) =>
+                          '{
+                            sangria.schema.Argument.createWithDefault(
+                              ${ Expr(name) },
+                              sangria.schema.OptionInputType(${ fieldType.asExprOf[InputType[i]] }),
+                              ${ flattenOptionExpr[String](description) },
+                              ${ defaultValue.asExprOf[d] }
+                            )($toInput, ${ fromInput }, ${ argType })
+                          }
+                        case optionTuple =>
+                          reportSummoningErrors(
+                            Seq(
+                              s"GraphQlInputType not found: ${TypeRepr.of[ToInput[d, _]].show}",
+                              s"GraphQlInputType not found: ${TypeRepr.of[FromInput[Option[i]]].show}",
+                              s"GraphQlInputType not found: ${TypeRepr.of[ArgumentType[Option[i]]].show}"
+                            ),
+                            optionTuple.toList
+                          )
+                case None =>
+                  (Expr.summon[FromInput[i]], Expr.summon[ArgumentType[i]]) match
+                    case (Some(fromInput), Some(argType)) =>
+                      '{
+                        sangria.schema.Argument.createWithoutDefault(
+                          ${ Expr(name) },
+                          ${ fieldType.asExprOf[InputType[i]] },
+                          ${ flattenOptionExpr[String](description) })($fromInput, $argType)
+                      }
+                    case optionTuple =>
+                      reportSummoningErrors(
+                        Seq(
+                          s"GraphQlInputType not found: ${TypeRepr.of[FromInput[i]].show}",
+                          s"GraphQlInputType not found: ${TypeRepr.of[ArgumentType[i]].show}"
+                        ),
+                        optionTuple.toList
+                      )
+              }
+          }
+        }
+
+        val optional = default.isEmpty && isSupertype[Option[_]](tpe)
+
+        val targetType =
+          if (optional)
+            tpe.baseType(TypeRepr.of[Option[_]].typeSymbol).typeArgs.head
+          else
+            tpe
+
+        NormalArg(name, targetType, expr, optional)
+    }
+
+  private def findKnownMembers[T](tpe: Type[T], includeMethods: Set[String]): List[KnownMember] =
+    import globalQuotes.reflect._
+    given Type[T] = tpe
+    val sym = TypeRepr.of[T].typeSymbol
+    (sym.fieldMembers ++ sym.methodMembers).flatMap { m =>
```

Comment:
The problem lies here, in Scala 2 version of the macro field members and method members are both obtained through one method, here as far as I know that is not possible so we end up messing with the order just by concatenating symbols like this. Moreover, even in tests which used only fields (or only methods), I could not always match the ordering, even reversing this list. I am unsure if if Scala 3 has their own guarantees for ordering as part of those methods, but I would imagine so

Reply:
Thanks for the info!
I suggest merging this PR, and looking back at the ordering if it's an issue later.